charts/tempo-distributed/values.yaml

```
@@ -1965,6 +1965,8 @@ gateway:
   extraVolumes: []
   # -- Volume mounts to add to the gateway pods
   extraVolumeMounts: []
+  # -- Enables the routes and ports that enable pushing traces using gRPC
+  enableGrpc: false
```

Wrapping the gRPC feature with `enableGrpc` flag looks good! however with the introduction of a new flag, confusion might arise among the users, therefore I would strongly recommend reverting to  the existing [`.Values.traces.otlp.grpc.enabled`](https://github.com/grafana/helm-charts/blob/ff6f9facdbf73c56e4bd09a32fce5705c934be6e/charts/tempo-distributed/values.yaml#L1210) boolean flag in order to extend this feature.

---

Considering the Helm Charts/Kubernetes best practices it would be much more clean, modular and flexible to introduce the default grpc port within [`Values.traces.otlp.grpc`](https://github.com/grafana/helm-charts/blob/ff6f9facdbf73c56e4bd09a32fce5705c934be6e/charts/tempo-distributed/values.yaml#L1208-L1212) block (making it configurable), where the port could then be referenced in the required templates.

Define the default gRPC port [here](https://github.com/grafana/helm-charts/blob/ff6f9facdbf73c56e4bd09a32fce5705c934be6e/charts/tempo-distributed/values.yaml#L1208-L1212) as in below suggestion
```suggestion
  grpc:
      # -- Enable Tempo to ingest Open Telemetry GRPC traces
      enabled: true
      # -- Default OTLP gRPC port
      port: 4317
      # -- GRPC receiver advanced config
      receiverConfig: {}
```